dashboard/src/components/common/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,156 @@
+   * Props for component slots.
+   */
+  slotProps?: {
+    /**
+     * Props to be passed to the next button component.
+     */
+    nextButton?: Partial<ButtonProps>;
+    /**
+     * Props to be passed to the previous button component.
+     */
+    prevButton?: Partial<ButtonProps>;
+  };
+};
+
+export default function Pagination({
+  className,
+  totalNrOfPages,
+  currentPageNumber,
+  onPrevPageClick,
+  onNextPageClick,
+  slotProps,
+  elementsPerPage,
+  onPageChange,
+  totalNrOfElements,
+  ...props
+}: PaginationProps) {
+  return (
+    <div
+      className={twMerge('grid grid-flow-col items-center gap-2', className)}
+      {...props}
+    >
+      <div className="grid justify-start grid-flow-col gap-2">
+        <Button
+          variant="outlined"
+          color="secondary"
+          className="block text-xs"
+          disabled={currentPageNumber === 1}
+          aria-label="Previous page"
+          onClick={onPrevPageClick}
+        >
+          <ChevronLeftIcon className="w-4 h-4" />
+          Back
+        </Button>
+
+        <div className="grid grid-cols-3 gap-1 text-center grid-col">
+          <Text className="self-center text-xs align-middle text-greyscaleGreyDark w-">
+            Page
+          </Text>
+          <Input
+            value={currentPageNumber}
+            onChange={(e) => {
+              const page = parseInt(e.target.value, 10);
+              if (page > 0 && page <= totalNrOfPages) {
+                onPageChange(page);
+              }
+            }}
+            disabled={totalNrOfPages === 1}
+            color="secondary"
+            componentsProps={{
+              formControl: {
+                className: 'flex flex-row-reverse text-center',
+              },
+              inputRoot: {
+                className: 'px-12',
+              },
+            }}
+            slotProps={{
+              root: {
+                className: twMerge(
+                  'w-9 h-8 text-center px-0.5',
+                  currentPageNumber > 9 ? 'px-0' : 'px-0.5',
+                ),
+              },
+              input: {
+                style: {
+                  fontSize: '0.70rem',
+                },
+              },
+            }}
+          />
+          <Text className="self-center text-xs align-middle text-greyscaleGreyDark">
+            of {totalNrOfPages}
+          </Text>
+        </div>
+
+        <Button
+          variant="outlined"
+          color="secondary"
+          className="text-xs"
+          aria-label="Next page"
+          disabled={currentPageNumber === totalNrOfPages}
+          onClick={onNextPageClick}
+          {...slotProps?.nextButton}
+        >
+          Next
+          <ChevronRightIcon className="w-4 h-4" />
+        </Button>
+      </div>
+      <div className="flex flex-row items-center justify-end text-center gap-x-1">
+        <Text className="text-xs text-greyscaleGreyDark">
+          {currentPageNumber === 1 && currentPageNumber}
```

Comment:
the second flag here, `currentPageNumber`, is redundant. Or maybe you wanted to use another flag.

Reply:
This is correct I think: given that the left hand contains an equals operator it's converted to a condition -- we want to evaluate it in order to show the right-hand (which just happens to be the same one as used in the condition) removing the second one would result in the expression being a boolean and not a string

I could change it for `{currentPageNumber === 1 && 1}` for clarity, adding to the expression below (or abstract it to a function) if it's easier to understand 👍